server/src/main/java/org/opensearch/index/shard/IndexShard.java

```diff
@@ -2234,6 +2236,10 @@ private void innerOpenEngineAndTranslog(LongSupplier globalCheckpointSupplier) t
             if (indexSettings.isRemoteStoreEnabled()) {
                 syncSegmentsFromRemoteSegmentStore(false);
             }
+            if (indexSettings.isRemoteTranslogStoreEnabled() && shardRouting.primary()) {
+                syncTranslogFilesFromRemoteTranslog();
+                loadGlobalCheckpointToReplicationTracker();
+            }
```

Comment:
can we extract this to a method - i see 2 separate occurrences in this files atleast. If there are more, we can just hit a method rather duplicating code.